commit 3

IsaevMilan opened the pull request:
Д.З. спринта 9

IsaevMilan committed:
commit 2

IsaevMilan committed:
commit 3.1

IsaevMilan committed:
commit 4 домашняя работа по 9 спринту

IsaevMilan committed:
commit 4.1 домашняя работа по 9 спринту
добавил файлы со шрифтами.

faritowich commented on app/src/main/res/values/dimens.xml:
```diff
@@ -4,4 +4,5 @@
     <dimen name="buttomHeight">220</dimen>
     <dimen name="textSizePlaylist">22sp</dimen>
     <dimen name="settingTextSize">16sp</dimen>
+    <dimen name="arrowBack">16</dimen>
```
Не используется, также странно, что без размерностей

faritowich commented on app/src/main/java/com/example/myplaylistmaker/SettingsActivity.kt:
```diff
@@ -1,19 +1,51 @@
+        val backButt = findViewById<ImageView>(R.id.arrowBack)
 
         backButt.setOnClickListener { val searchIntent = Intent (this, MainActivity::class.java )
             startActivity(searchIntent)
```
1) Для перехода назад стоит использовать не интент, а методы finish() или onBackPressed(), потому что иначе вместо возврата в предыдущий экран, создается новый экземпляр MainActivity и при возврате обратно на главный экран и нажатии на кнопку телефона "назад" мы "вернемся" снова на экран настроек 

2) Немного не котлиновское форматирование. Есть хороший хоткей, можно нажать Alt + Ctrl + L и студия красиво отформатирует весь код на экране.

IsaevMilan replied:
понял, исправлю.

faritowich commented on app/src/main/java/com/example/myplaylistmaker/SettingsActivity.kt:
```diff
@@ -1,19 +1,51 @@
+            shareIntent.putExtra(Intent.EXTRA_SUBJECT, subject)
+            shareIntent.putExtra(Intent.EXTRA_TEXT, message)
+            startActivity(shareIntent)
+        }
```
Можно ещё использовать функции области видимости, чтобы такие фрагменты кода сокращать до:

```
            Intent(Intent.ACTION_SENDTO).apply {
                val subject = "Спасибо разработчикам и разработчицам приложения Playlist Maker!"
                val message = "Спасибо разработчикам и разработчицам за крутое приложение!"
                data = Uri.parse("mailto:")
                putExtra(Intent.EXTRA_EMAIL, arrayOf("YAmilanisaev@yandex.ru"))
                putExtra(Intent.EXTRA_SUBJECT, subject)
                putExtra(Intent.EXTRA_TEXT, message)
                startActivity(this)
            }

```
Конечно, с учетом выноса строк в ресурсы. 
Про такие методы почитать можно здесь:
https://kotlinlang.ru/docs/reference/scope-functions.html

IsaevMilan replied:
вот тут мне как новичку сложновато будет, с учетом того что я отстаю, если это не критично на данном этапе не стал бы в логике работающей что-то менять здесь, боюсь что-нибудь испортить))

faritowich replied:
Этот момент уже некритичный, дело вкуса, просто полезно знать о таких конструкциях, часто встречаются

faritowich commented on app/src/main/res/values/Styles.xml:
```diff
@@ -0,0 +1,69 @@
+       <item name="android:layout_width">match_parent</item>
+       <item name="android:layout_height">match_parent</item>
+       <item name="android:layout_marginTop">24dp</item>
+       <item name="android:layout_weight">1</item>
```
Лучше внешние отступы, высоту и ширину, веса не выносить в стиль. Если на экране будет много элементов, будет сложнее ориентироваться с их взаимным расположением, когда отвечающие за это атрибуты разнесены по разным файлам - что-то может быть в лейауте, что-то в стиле. В стилях обычно содержат атрибуты, отвечающие за внешний вид вьюхи - цвета, шрифты, фоны и т.д. Бывают исключения, когда, например, мы точно знаем, что у кнопки всегда должна строго определенная высота и ширина - тогда да, имеет смысл выносить

Также само название файла styles принято писать с маленькой буквы (см. остальные файлы в res)

IsaevMilan replied:
Понял исправлюсь

faritowich commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -1,38 +1,69 @@
     android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_height="wrap_content"
```
Лучше match_parent. Корневой лейаут чаще всего должен заполнить весь экран, если он ему соответствует.

IsaevMilan replied:
ок

faritowich commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -1,38 +1,69 @@
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_gravity="end|center_vertical"
+            android:layout_marginEnd="13dp"
```
Рекомендуется делать размеры кратными 8 (или 4). Это обеспечивает единообразие дизайна, а также связано с тем, что большинство размеров экрана делятся на 8. Иногда эту рекомендацию приходится нарушать, но лучше по возможности ей следовать.
Также см. https://material.io/design/layout/spacing-methods.html#baseline-grid

IsaevMilan replied:
о дельный совет, не знал спасибо)

faritowich commented on app/src/main/java/com/example/myplaylistmaker/SearchActivity.kt:
```diff
@@ -1,22 +1,90 @@
+        const val INPUT_TEXT = "input_text"
+    }
+
+    private var saveText=("")
```
Можно и даже лучше без скобок

IsaevMilan replied:
ок)

faritowich commented on app/src/main/java/com/example/myplaylistmaker/SearchActivity.kt:
```diff
@@ -1,22 +1,90 @@
 class SearchActivity : AppCompatActivity() {
     @SuppressLint("MissingInflatedId")
+    companion object {
+        const val INPUT_TEXT = "input_text"
```
Если используем только в этом классе, то private

faritowich commented on app/src/main/java/com/example/myplaylistmaker/SearchActivity.kt:
```diff
@@ -1,22 +1,90 @@
-        backSbutton.setOnClickListener { val settingsIntent = Intent (this, MainActivity::class.java )
-            startActivity(settingsIntent)
+        backButt.setOnClickListener { val searchIntent = Intent (this, MainActivity::class.java )
+            startActivity(searchIntent)
```
См. похожий коммент по экрану настроек

IsaevMilan committed:
commit 4.2 домашняя работа по 9 спринту.
Исправлены критические и рекомендованные ошибки.
!!!Я не нашел правда тексты не вынесенные в ресурсы.!!!

IsaevMilan committed:
commit 4.4

IsaevMilan committed:
commit 4.4 final

IsaevMilan committed:
commit 5.1 final Домашка спринт 10.

faritowich commented on app/src/main/java/com/example/myplaylistmaker/SearchActivity.kt:
```diff
@@ -1,22 +1,92 @@
+            inputEditText.clearFocus()
+        }
+
+    val simpleTextWatcher = object : TextWatcher {
```
Немного поплыло форматирование. Есть хороший хоткей, можно нажать Alt + Ctrl + L и студия красиво отформатирует весь код на экране

faritowich commented on app/src/main/java/com/example/myplaylistmaker/MediaAdapter.kt:
```diff
@@ -0,0 +1,25 @@
+
+    override fun getItemCount(): Int {
+        return track.size
+    }
```
Можно также сократить до `override fun getItemCount() = track.size`

faritowich commented on app/src/main/java/com/example/myplaylistmaker/SearchActivity.kt:
```diff
@@ -1,22 +1,92 @@
+            )
+        )
+
+        val rvTrack = findViewById<RecyclerView>(R.id.rvTracks)
```
Можно также объявлять переменные, отвечающие за View, в теле класса с lateinit (будет private lateinit var view: View) или как нуллабл переменную (будет private var view: View? = null, (используется реже)), а в onCreate уже задавать значение findViewById(R.id.viewId). Все остальные переменные с UI можно объявить так же
Это позволит использовать переменные во всем классе и не объявлять перменные, отвечающие за View, в разных местах класса

IsaevMilan committed:
commit 5.1 final Домашка спринт 10.

IsaevMilan committed:
commit 6.1  Домашка спринт 11.

IsaevMilan committed:
commit 6.1  Домашка спринт 11.

IsaevMilan committed:
commit 6.2  Домашка спринт 11.

gusar-off commented on app/src/main/java/com/example/myplaylistmaker/ApiInterface.kt:
```diff
@@ -0,0 +1,23 @@
+
+
+
+
```
Лишние отступы, можно убрать

gusar-off commented on app/src/main/java/com/example/myplaylistmaker/MediaResponse.kt:
```diff
@@ -0,0 +1,15 @@
+
+
+
+     
```
и тут

gusar-off commented on app/src/main/java/com/example/myplaylistmaker/MediaViewHolder.kt:
```diff
@@ -0,0 +1,35 @@
+import com.bumptech.glide.Glide
+import com.bumptech.glide.load.resource.bitmap.RoundedCorners
+import java.text.SimpleDateFormat
+import java.util.*
```
Wild card import-ы не советуют использовать в код стилях Kotlin. В настройках Android студии можно отключить https://developer.android.com/kotlin/style-guide#import_statements

gusar-off commented on app/src/main/java/com/example/myplaylistmaker/SearchActivity.kt:
```diff
@@ -1,22 +1,202 @@
\ No newline at end of file
+
+
+//    private fun showMessage(text: FrameLayout, additionalMessage: String) {
```
Закомментированный код лучше убрать.

gusar-off commented on app/src/main/java/com/example/myplaylistmaker/SearchActivity.kt:
```diff
@@ -1,22 +1,202 @@
+                            } else {
+                                media.clear()
+                                moviesList.visibility = View.GONE
+                                placeholderLineErr.visibility = View.VISIBLE
```
Чтобы не дублировать, можно вынести в метод работу с видимостью элементов

IsaevMilan committed:
commit 7  Домашка промежуточная спринт 12.

IsaevMilan committed:
commit 8  Домашка  спринт 12.

IsaevMilan committed:
commit 9  Домашка  спринт 12.

IsaevMilan committed:
commit 10  Домашка  спринт 12.

IsaevMilan committed:
commit 11  Домашка  спринт 12.

IsaevMilan committed:
commit 12  Домашка  спринт 12.

IsaevMilan committed:
commit 13  Домашка  спринт 12.

IsaevMilan committed:
commit 14  Домашка  спринт 12.

IsaevMilan committed:
commit 15  Домашка  спринт 12.
